llvm/include/llvm/CodeGen/GlobalISel/GIMatchTableExecutorImpl.h

```diff
@@ -1354,8 +1385,8 @@ bool GIMatchTableExecutor::executeMatchTable(
     }
 
     case GIR_MakeTempReg: {
-      int64_t TempRegID = MatchTable[CurrentIdx++];
-      int64_t TypeID = MatchTable[CurrentIdx++];
+      uint64_t TempRegID = nextULEB128();
+      int64_t TypeID = (int8_t)MatchTable[CurrentIdx++];
```

Comment:
This casting is weird.

Reply:
Indeed but it's needed, I've added a `readS8` wrapper around it with a comment explaining why we use that.